pkg/dialer/dialer_windows.go

```diff
@@ -37,7 +39,33 @@ func isNoent(err error) bool {
 }
 
 func dialer(address string, timeout time.Duration) (net.Conn, error) {
-	return winio.DialPipe(address, &timeout)
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	// If there is nobody serving the pipe we limit the timeout for this case to
+	// 5 seconds because any shim that would serve this endpoint should serve it
+	// within 5 seconds.
+	serveTimer := time.NewTimer(5 * time.Second)
```

Comment:
Would be good to get some feedback from one of the Windows developers on this--this comment and the timeout seems to be specific to shims, but now is in the code used by many different parts of the code (Go client, server-side gRPC connection(s), etc.). It also seems like a lot of timeouts going on.. the timeout from the caller; if you look at the wrapping call it also has a timeout running while calling the dialer func. Would be good to get clarity, and at a minimum if there is specific shim-behavior then there probably has to be a wrapping call for shims to use separately from the unified dialer function.

Reply:
@estesp - For `RuntimeV2` (shim) activation's Windows uses named pipes not a fifo. Due to this client server model it means that there is a delay between when the call to `shim.exe start ...` and the actual pipe is served by the shim. This timeout is meant to handle that case but I agree with such a generic dialer it doesn't make a ton of sense. We really should have that caller pass in the increased timeout knowing that its a `shim.exe start ...` call.

Reply:
It also seems weird that the dialer package already handles retries every 10ms up to timeout and this `dialer` func will do the same

Reply:
@jterry75 let me see if I get this correctly, because I really have no idea how any of this works on windows:

- `pkg/dialer/dialer_windows.go` should just do ` winio.DialPipe(address, &timeout)` as before this PR and ttrpcutil should just call it

- this awkward 5 second retry logic should only be left inside the runtimeV2's shim

Reply:
Yes I think that is correct. It should issues a `DialPipe` as it did previously and only handle `os.IsNotExist` as a retry candidate in `isNoent`. Otherwise it should return the error immediately. The caller from runtime v2 shim should use the 5 second timeout